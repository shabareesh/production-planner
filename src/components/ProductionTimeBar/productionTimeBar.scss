.production-time-bar {
  display: flex;
  text-align: center;
  font-size: 12px;
  font-weight: 700;
  line-height: 15px;
  color: #fff;

  &--prepared {
    background: #F7C136;
    border-radius: 2px;
  }

  &--running {
    background: #31AD2E;
    border-radius: 2px;
  }

  &--pending {
    background: #9AD296;
    border-radius: 2px;
  }
}
