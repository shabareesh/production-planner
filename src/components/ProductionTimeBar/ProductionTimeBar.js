import React from 'react';
import PropTypes from 'prop-types';

import './productionTimeBar.scss';
import {calculatePercent} from "../../common/utils";

const ProductionTimeBar = ({ totalTime, preparedTime, runningTime }) => (
    <div className="production-time-bar">
        <span
            className="production-time-bar--prepared"
            style={{width: `${calculatePercent(totalTime, runningTime)}%`}}
        >
            {preparedTime}
        </span>
        <span
            className="production-time-bar--running"
            style={{width: `${calculatePercent(totalTime, runningTime)}%`}}
        >
            {runningTime}
        </span>
        <span
            className="production-time-bar--pending"
            style={{width: `${calculatePercent(totalTime, totalTime - preparedTime - runningTime)}%`}}
        >
            {totalTime - preparedTime - runningTime}
        </span>
    </div>
);

ProductionTimeBar.propTypes = {
    totalTime: PropTypes.number.isRequired,
    preparedTime: PropTypes.number.isRequired,
    runningTime: PropTypes.number.isRequired,
};

export default ProductionTimeBar;
