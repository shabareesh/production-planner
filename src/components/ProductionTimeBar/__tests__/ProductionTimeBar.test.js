import React from 'react';
import {render} from "@testing-library/react";

import ProductionTimeBar from "../ProductionTimeBar";

describe('ProductionTimeBar', () => {
   test('should match snapshot', () => {
      const { baseElement } = render(<ProductionTimeBar preparedTime={2} runningTime={2} totalTime={12} />);
      expect(baseElement).toMatchSnapshot();
   });
});
