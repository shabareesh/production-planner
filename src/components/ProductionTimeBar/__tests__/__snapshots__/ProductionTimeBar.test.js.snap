// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductionTimeBar should match snapshot 1`] = `
<body>
  <div>
    <div
      class="production-time-bar"
    >
      <span
        class="production-time-bar--prepared"
        style="width: 16.67%;"
      >
        2
      </span>
      <span
        class="production-time-bar--running"
        style="width: 16.67%;"
      >
        2
      </span>
      <span
        class="production-time-bar--pending"
        style="width: 66.67%;"
      >
        8
      </span>
    </div>
  </div>
</body>
`;
